ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+# #
+# Software subject to following license(s):
+#   Apache 2 License (http://www.opensource.org/licenses/apache2.0)
+#   Copyright (c) Responsible Organization
+#
+
+# #
+# Current developer(s):
+#   Jindrich Novy <jindrich.novy@morganstanley.com>
+#
+
+# #
+# Author(s): Jindrich Novy
+#
+
+package NCM::Component::spma::yumng;
+
+#
+# a few standard statements, mandatory for all components
+#
+use strict;
+use warnings;
+use NCM::Component;
+our $EC  = LC::Exception::Context->new->will_store_all;
+our @ISA = qw(NCM::Component);
+use EDG::WP4::CCM::Element qw(unescape);
+
+use CAF::Process;
+use CAF::FileWriter;
+use Set::Scalar;
+use File::Path qw(mkpath rmtree);
+use LWP::Simple;
+use Text::Glob qw(match_glob);
+
+use constant HOSTNAME            => "/system/network/hostname";
+use constant DOMAINNAME          => "/system/network/domainname";
+use constant REPOS_DIR           => "/etc/yum.repos.d";
+use constant REPOS_TREE          => "/software/repositories";
+use constant PKGS_TREE           => "/software/packages";
+use constant GROUPS_TREE         => "/software/groups/names";
+use constant CMP_TREE            => "/software/components/spma";
+use constant YUM_PACKAGE_LIST    => "/etc/yum/pluginconf.d/versionlock.list";
+use constant YUM_CONF_FILE       => "/etc/yum.conf";
+use constant RPM_QUERY_INSTALLED => qw(rpm -qa --nosignature --nodigest --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH}\n);
+use constant REPO_AVAIL_PKGS     => qw(repoquery -C --show-duplicates --all --qf %{EPOCH}:%{NAME}-%{VERSION}-%{RELEASE}.%{ARCH});
+use constant YUM_PLUGIN_OPTS	 => "--disableplugin=\* --enableplugin=fastestmirror --enableplugin=versionlock";
+use constant YUM_TEST_CHROOT     => qw(/tmp/spma_yum_testroot);
+use constant SPMA                => "/software/components/spma";
```

Comment:
In fact $self->prefix() doesn't work, so I will stick with the static path:

[INFO] Errors while configuring spma (1)
[ERROR] 1 errors, 0 warnings executing configure
[ERROR] Uncaught exception!
[ERROR] Calling stack is:
[ERROR] *** cannot resolve element /software/components/yumng

[ERROR] Uncaught exception!
[ERROR] Calling stack is:
        LC::Exception::throw_error called at /opt/edg/lib/perl/EDG/WP4/CCM/Configuration.pm line 249

        EDG::WP4::CCM::Configuration::getElement called at /opt/edg/lib/perl/NCM/Component/spma/yumng.pm line 285

        NCM::Component::spma::yumng::Configure called at /opt/edg/lib/perl/NCM/Component/spma.pm line 74

        NCM::Component::spma::call_entry_point called at /opt/edg/lib/perl/NCM/Component/spma.pm line 33

        NCM::Component::spma::Configure called at (eval 20) line 1

        (eval) called at /opt/edg/lib/perl/NCD/ComponentProxy.pm line 514

        NCD::ComponentProxy::_execute called at /opt/edg/lib/perl/NCD/ComponentProxy.pm line 94

        NCD::ComponentProxy::executeConfigure called at /opt/edg/lib/perl/NCD/ComponentProxyList.pm line 146

        NCD::ComponentProxyList::executeConfigComponents called at /usr/sbin/ncm-ncd line 465

[ERROR] *** EDG::WP4::CCM::Element->createElement (EDG::WP4::CCM::Configuration=HASH(0x2057deb0), EDG::WP4::CCM::Path=ARRAY(0x20dfb220)): Failed to read type of /software/components/yumng